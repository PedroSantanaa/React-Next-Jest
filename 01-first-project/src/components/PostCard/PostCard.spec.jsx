const { render, screen } = require('@testing-library/react');
const { PostCard } = require('.');
const { PostCardPropsMock } = require('./mock');

const props = PostCardPropsMock;

describe('<PostCard/>', () => {
  it('should render PostCArd correctly', () => {
    // render(<PostCard title="algo" cover="url" body="Some text" id="1" />);
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: /TITLE1/i })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: 'title1 1' })).toBeInTheDocument();
    expect(screen.getByText('body1')).toBeInTheDocument();
  });
  it('should match the snapshot', () => {
    // render(<PostCard title="algo" cover="url" body="Some text" id="1" />);
    const { container } = render(<PostCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
