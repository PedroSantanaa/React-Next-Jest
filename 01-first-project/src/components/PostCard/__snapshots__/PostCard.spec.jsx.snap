// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match the snapshot 1`] = `
<div>
  <div
    class="post"
  >
    <img
      alt="title1"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title1
         
        1
      </h2>
      <p>
        body1
      </p>
    </div>
  </div>
</div>
`;
